hermes/bin/src/runtime_extensions/hermes/cron/state.rs

```diff
@@ -0,0 +1,490 @@
+//! Cron State.
+/// The crontabs hash map.
+use std::hash::{Hash, Hasher};
+
+use once_cell::sync::Lazy;
+use tokio::{
+    runtime::Builder,
+    sync::{mpsc, oneshot},
+};
+
+use super::{
+    event::OnCronEvent,
+    queue::{CronEventQueue, CronJob, CronJobDelay},
+};
+use crate::{
+    app::HermesAppName,
+    event::{queue::send, HermesEvent, TargetApp, TargetModule},
+    runtime_extensions::{
+        bindings::hermes::cron::api::{CronEventTag, CronTagged, Instant},
+        hermes::cron::mkdelay_crontab,
+    },
+};
+
+/// Cron Internal State
+static CRON_INTERNAL_STATE: Lazy<InternalState> = Lazy::new(|| {
+    let sender = if let Ok(runtime) = Builder::new_current_thread().enable_all().build() {
```

Comment:
What do you think to make this logic as a part of the `HermesReactor` so it will be possible to maintain this execution loop of the cron events ?
Same as it was done for the hermes event queue

Reply:
Actually... thinking a bit more about this, I think that we should keep the join handle and make it more maintainable, but lets talk about how coupled the HermesReactor should be to the module APIs to discuss the best approach.